lib/mayaUsd/fileio/shading/shadingModeUseRegistry.cpp

```diff
@@ -271,17 +271,16 @@ class UseRegistryShadingModeExporter : public UsdMayaShadingModeExporter
 
                 // We pass in the type of the plug on the other side to allow the export code to
                 // add conversion nodes as required.
-
-                const TfToken srcPlugName
-                    = TfToken(UsdMayaShadingUtil::GetStandardAttrName(srcPlug, false));
-                UsdAttribute srcAttribute = srcShaderInfo->GetShadingAttributeForMayaAttrName(
-                    srcPlugName, MayaUsd::Converter::getUsdTypeName(dstPlug));
-
                 const TfToken dstPlugName
                     = TfToken(UsdMayaShadingUtil::GetStandardAttrName(dstPlug, false));
                 UsdAttribute dstAttribute = dstShaderInfo->GetShadingAttributeForMayaAttrName(
                     dstPlugName, MayaUsd::Converter::getUsdTypeName(srcPlug));
 
+                const TfToken srcPlugName
+                    = TfToken(UsdMayaShadingUtil::GetStandardAttrName(srcPlug, false));
+                UsdAttribute srcAttribute = srcShaderInfo->GetShadingAttributeForMayaAttrName(
+                    srcPlugName, dstAttribute.GetTypeName());
```

Comment:
The src will often create an output to match the requested input type, so be extremely precise by returning the expected USD attribute type instead of the Maya type of the connection. This allows converting an outColor connection into a connection to a subcomponent.